### PR TITLE
Improves GAME's scoring module, mostly from the following two perspec…

### PR DESCRIPTION
**Pull request by XianXing**
…tives: 1. Separate the logic of scoring and model evaluation 2. Created a more user-friendly output of the scoring job. Also improved test coverage for scoring related components.

More details can be found in issue #71.

### CODE REVIEW

File: photon-ml/src/main/scala/com/linkedin/photon/ml/cli/game/scoring/Params.scala [hunk 1]
```diff
@@ -60,49 +81,57 @@ object Params {
...
+              s"e.g. 90-1. If this parameter is specified, the input directory is expected to be in the " +
+              s"daily format structure (e.g., inputDir/daily/2015/05/20/input-data-files). Otherwise, the input paths " +
+              s"are assumed to be flat directories of input files (e.g., inputDir/input-data-files). " +
+              s"Default: ${defaultParams.dateRangeDaysAgoOpt}.")
+          .action((x, c) => c.copy(dateRangeDaysAgoOpt = Some(x)))
         .action((x, c) => c.copy(dateRangeOpt = Some(x)))
       opt[String]("feature-name-and-term-set-path")
           .required()
-          .text("Input path to the features name-and-term lists.")
+          .text(s"Input path to the features name-and-term lists.")
```
Comment (ashelkovnykov) [inline photon-ml/src/main/scala/com/linkedin/photon/ml/cli/game/scoring/Params.scala]: Are the 's' characters necessary at the start of the strings even if no string interpolation is happening?

File: photon-ml/src/main/scala/com/linkedin/photon/ml/cli/game/scoring/Driver.scala [hunk 2]
```diff
@@ -44,6 +42,7 @@ class Driver(val params: Params, val sparkContext: SparkContext, val logger: Pho
     s"${sparkContext.getExecutorStorageStatus.length * 3}").toInt
   protected val hadoopConfiguration = sparkContext.hadoopConfiguration
 
+  //TODO: There might be some potential inconsistency when the data
```
Comment (ashelkovnykov) [inline photon-ml/src/main/scala/com/linkedin/photon/ml/cli/game/scoring/Driver.scala]: I believe there's a typo here

**Comment by ashelkovnykov** [comment]
LGTM, I actually made some of the same changes you did while cleaning up for the OptimizationProblem synthesis.

File: photon-ml/src/main/scala/com/linkedin/photon/ml/cli/game/scoring/Driver.scala (hunk 2 above)
Comment (XianXing) [inline photon-ml/src/main/scala/com/linkedin/photon/ml/cli/game/scoring/Driver.scala]: Let me try to fix it and please let me know if it works or not :)

File: photon-ml/src/main/scala/com/linkedin/photon/ml/cli/game/scoring/Params.scala (hunk 1 above)
Comment (XianXing) [inline photon-ml/src/main/scala/com/linkedin/photon/ml/cli/game/scoring/Params.scala]: No they are redundant. Will fix it.

**Comment by joshvfleming** [comment]
LGTM

**Pull request merged**